docs/languages/csharp/expression-trees-building.md

```diff
@@ -0,0 +1,218 @@
+local variables before you use them. Once you've created those objects,
+you can use them in your expression tree wherever you need.
+
+Second, you need to use a subset of the Reflection APIs to create a `MethodInfo` object
+so that you can create an expression tree to access that method. You must limit
+yourself to the subset of the Reflection APIs that are available on the .NET Core platform. Again,
+these techniques will extend to other expression trees. You'll have to create a
+delegate that cals the method you want, and retrieve the `MethodInfo` object
+from that expression.
+
+## Building Code In Depth
+
+You aren't limited in what you can build using these APIs. However, the more
+complicated expression tree that you want to build, the more difficult
+the code is to manage and to read. 
+
+Let's build an expression tree that is the equivalent of this code:
+
+```cs
+Func<int, int> factorialFunc = (n) =>
+{
+    var res = 1;
+    while (n > 1)
+    {
+        res = res * n;
+        n--;
+    }
+    return res;
+};
+```
+
+Notice above that I did not build the expression tree, but simply the delegate. Using
+the `Expression` class, you can't build statement lambdas. Here's the code that is required
+to build the same functionality. It's complicated by the fact that there isn't an API to build
+a `while` loop, instead you need to build a loop that contains a conditional test, and a label
+target to break out of the loop. 
+
+```cs
+var nArgument = Expression.Parameter(typeof(int), "n");
+var result = Expression.Variable(typeof(int), "result");
+
+// Creating a label that represents the return value
+LabelTarget label = Expression.Label(typeof(int));
+
+var initializeResult = Expression.Assign(result, Expression.Constant(1));
+
+// This is the inner block that performs the multiplication,
+// and decrements the value of 'n'
+var block = Expression.Block(
+    Expression.Assign(result,
+        Expression.Multiply(result, nArgument)),
+    Expression.PostDecrementAssign(nArgument)
+);
+
+// Creating a method body.
+BlockExpression body= Expression.Block(
+    new[] { result },
+        initializeResult,
+        Expression.Loop(
+            Expression.IfThenElse(
+                Expression.GreaterThan(nArgument, Expression.Constant(1)),
+                block,
+                Expression.Break(label, result)
+            ),
+        label
+    )
+);
+```
+
+The code to build the code is quite a bit longer, more complicated, and it's riddled
+with labels and gotos and other elements we'd like to avoid in our everyday
+coding tasks. 
+
+For this section, I've also updated the visitor code to visit every node in this expression
+tree and write out information about the nodes that are created in this sample. You can see
+the code in the samples section. You can experiment for yourself: build it and run the samples.
+
+## Examining the APIs
+
+The expression tree APIs are some of the more difficult to navigate in the
```

Comment:
I'd rewrite this as "[...] some of the more difficult APIs to navigate in .NET Core.  But that's fine!  [...]"